entity-service-impl/src/main/java/org/hypertrace/entity/query/service/EntityQueryServiceImpl.java

```diff
@@ -564,6 +570,12 @@ private List<Document> getProjectedDocuments(
   private void doBulkUpdate(
       RequestContext requestContext, String entityType, Map<String, EntityUpdateInfo> entitiesMap)
       throws Exception {
+    Optional<String> maybeTenantId = requestContext.getTenantId();
+    if (maybeTenantId.isEmpty()) {
+      return;
```

Comment:
Will we throw an exception eventually to end API caller?

Reply:
The caller already checks the presence of tenant id in request context, and returns an error

Reply:
In that case, could have simplified to
```
String tenantId = requestContext.getTenantId().orElseThrow();
```

Just by looking at the `doBulkUpdate()` method, it feels like it is a no-op when the tenantId is missing. Throwing exception would improve the readability like `tenantId` is expected to be present otherwise the operation would fail.

Reply:
If a caller is making sure, we don't need these additional checks.